File: Computer architecture work/Homework 1/homework1-Rickyocon-main/src/Computer Arch HW 1/Part2/SwapUsingTemp.c
//Method to swap using a temp variable


#include <stdio.h>
//library to get random number
#include <stdlib.h>
//Library to calculate time
#include <time.h>

//Method to swap using a temp variable
int SwapUsingTemp()
{
    //initilize counter  
    int counter = 0;
    
    //loop 1000 times
    for(int i=0;i<1000;i++)
    {
      //Get random value for x and y
      int x = rand()%1000;
      int y = rand()%1000;
      int temp;
      
      //Print values of x and y before swap
      printf("Before swaping: x = %d and y = %d",x,y);
      printf("\n");
      
      //Swap using temp variable
      temp = x;
      x = y;
      y = temp;
      counter += 1;
      
      //print values of x and y after swaping
      printf("After swaping: x = %d and y = %d",x,y);
      printf("\n");
      printf("\n");
      
      
    }
    
    //print how many times the loop has run
    printf("This loop has run %d times", counter);
    
    
}



int main()

{
    // To calculate how long the program took
    clock_t t;
    t = clock();
    SwapUsingTemp();
    t = clock() - t; 
    double time_taken = ((double)t)/CLOCKS_PER_SEC;
    printf("\n");
    printf("This program took %f seconds to execute \n", time_taken);
  
  
}

File: Computer architecture work/Homework 1/homework1-Rickyocon-main/src/Computer Arch HW 1/Part2/SwapUsingTemp.java
// Java program to swap two variables
// Using a Temp variable

public class SwapUsingTemp 
{

	public static void main(String[] args)
	{    
     
        //Start timer
        long StartTime = System.nanoTime(); 
    
        //initilize counter
        int counter = 0;

        //loop 1000 times
        for(int i = 1;i<=1000;i++)
        {
            //Get random value for x and y
            double x = (Math.random()*1000);
            double y = (Math.random()*1000);

            //Print values of x and y before swap
            System.out.println("Before Swap");
		    System.out.println("x = " + x);
		    System.out.println("y = " + y);

            //swap using temp variable
		    double temp = x;
		    x = y;
		    y = temp;
            counter += 1;

            //print values of x and y after swaping
		    System.out.println("After swap");
		    System.out.println("x = " + x);
		    System.out.println("y = " + y);
            //print how many times the loop has run
            System.out.println("The loop has run " +counter +" times");

        }

        //end timer and calculate elapsed time
        long duration = (System.nanoTime() - StartTime)/1000000;  //Calculates time in mili seconds
        //print duration
        System.out.println("The program took " +duration + " ms");

	}
}



